cmd/analyzer/analyzer.go

```diff
@@ -239,12 +241,37 @@ func NewService(cfg *config.AnalysisConfig) (*Service, error) {
 func (a *Service) Start() {
 	a.logger.Info("starting analysis service")
 
+	var ctx context.Context
+	ctx, a.cancelAnalyzers = context.WithCancel(context.Background())
+
+	// trap Ctrl+C and call cancel on the context
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan, os.Interrupt)
+	defer func() {
+		signal.Stop(signalChan) // Stop catching Ctrl+C signals.
+		a.Shutdown()            // Shut down gracefully.
+	}()
+
+	go func() {
+		select {
+		case <-signalChan:
+			// We received the first Ctrl+C; try to shut down gracefully.
+			a.Shutdown()
```

Comment:
is this to say we yank out the database connection on the first ctrl+c?

Reply:
Yes. And and the kv store.

Reply:
could we wait for the analyzers to return before doing that? this PR is about graceful shutdown, and this sounds like we won't really let analyzers finish what they're doing

Reply:
o true. gonna take a look, thanks

Reply:
:+1: light touch context cancellation :feather: